### ISSUE TITLE
aws_ami data source most_recent sorts for oldest instead of newest

### ISSUE THREAD

**Issue by danielnbarbosa** [hashicorp/terraform#9284]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
`Terraform v0.7.5`

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_ami

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
data "aws_ami" "backup_ami" {
  most_recent = true

  filter {
    name   = "name"
    values = ["${var.backup_ami_name_filter}"]
  }

  owners = ["${var.backup_ami_owner_id}"]
}
```

### Expected Behavior
AMI data source should return most recent version that matches the filter.

### Actual Behavior
AMI data source returned oldest version that matches the filter.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform plan`

**Comment by kwilczynski** [comment]
@danielnbarbosa hi there! I am sorry that you are having issues with this data source.

We have already addressed it in https://github.com/hashicorp/terraform/pull/9277.

Again, I am very sorry that you have problems.

**Comment by stack72** [comment]
Closed via #9277
